### PR TITLE
added data source for storage bucket object content

### PR DESCRIPTION
**Pull request by edwardmedia**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/3100

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
`google_storage_bucket_object_content`
```

### CODE REVIEW

File: third_party/terraform/website/docs/d/storage_bucket_object_content.html.markdown [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+page_title: "Google: google_storage_bucket_object_content"
+sidebar_current: "docs-google-datasource-storage-bucket-object_content"
+description: |-
+  Get content of a Google Cloud Storage bucket object.
+---
+
+
+# google\_storage\_bucket\_object\_content
+
+Gets an existing object content  inside an existing bucket in Google Cloud Storage service (GCS).
```
Comment (ScottSuarez) [inline third_party/terraform/website/docs/d/storage_bucket_object_content.html.markdown:13]:
```suggestion
Gets an existing object content inside an existing bucket in Google Cloud Storage service (GCS).
```

File: third_party/terraform/data_sources/data_source_storage_bucket_object_content.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"google.golang.org/api/storage/v1"
+)
+
+func dataSourceGoogleStorageBucketObjectContent() *schema.Resource {
+
+	dsSchema := datasourceSchemaFromResourceSchema(resourceStorageBucketObject().Schema)
+
+	addRequiredFieldsToSchema(dsSchema, "bucket")
+	addRequiredFieldsToSchema(dsSchema, "name")
+	addOptionalFieldsToSchema(dsSchema, "content")
```
Comment (ScottSuarez) [inline third_party/terraform/data_sources/data_source_storage_bucket_object_content.go:18]: Should this be an output only field?

**Review by ScottSuarez — APPROVED** [review]

File: third_party/terraform/data_sources/data_source_storage_bucket_object_content.go (hunk 2 above)
Comment (edwardmedia) [reply third_party/terraform/data_sources/data_source_storage_bucket_object_content.go:18]: It has been taken care of in https://github.com/GoogleCloudPlatform/magic-modules/blob/16c462951f174f835bb60bf481e8ec29278a3fb1/third_party/terraform/utils/datasource_helpers.go#L71:6

Comment (ScottSuarez) [reply third_party/terraform/data_sources/data_source_storage_bucket_object_content.go:18]: Computed is false here. Should be true ?

**Pull request merged**